### PR TITLE
Sync CURL_DISABLE options in CMake

### PR DESCRIPTION
**Pull request by donny-dont**
Adds the full listing of CURL_DISABLE options to the CMakeLists.txt. Moves all option code, except for CURL_DISABLE_OPENSSL_AUTO_LOAD_CONFIG which resides near OpenSSL configuration, to the same block of code. Also sorts the options.

Additionally sorted the CURL-DISABLE listing and fixed the CURL_DISABLE_POP3 option.

### CODE REVIEW

File: docs/CURL-DISABLE.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ Disable the auto load config support in the OpenSSL backend.
 
 Disable date parsing
 
-## CURL_DISABLE_POP
+## CURL_DISABLE_POP3
 
 Disable the POP(S) protocols
```
Comment (bagder) [inline docs/CURL-DISABLE.md:95]: Maybe fix this line too?

Comment (donny-dont) [reply docs/CURL-DISABLE.md:95]: Sure just change it to POP3 protocol?

**Comment by donny-dont** [comment]
Ok I think I got everything ready @bagder

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**